### ISSUE TITLE
Debricking Xiaomi Mi Router 4A Gigabit Edition

### ISSUE THREAD

**Issue by deathline94** [acecilia/OpenWRTInvasion#190]
Hey there, guys. The title is self-explanatory, but here it is
my router was shipped with the 3.2.30 firmware, and it was the English/global version. I tried to flash the latest OpenWrt, albeit for the v2 version, using the exploit.
But I guess my router was the v1 kind. So it never booted up again
orange light staying on.
It does go to the safe mod with a blinking orange light and when using the Tinypxe the light blinks fast and turns purple, but it stays that way and won't flash all the way through
so what are my options now ? Is it fixable ? What can I do ?
Thanks for the help in advance.

**Comment by nginx74** [comment]
Same happened for me.
the orange light keeps blinking, and I can't put it in boot mode.
Cant use TinyPXE for it, it does not respond to TinyPXE

**Comment by StefanStanic95** [comment]
Got the same problem on another router 4A Gigabit Global version router, fast blinking purple light that blinks infinitely. I am going to order a CH341A programmer and flash the chip from another 4A Gigabit global router which has the exact same firmware, 3.0.35 and report here if I succeed

In my case the router was bricked by:
- Flashing from 3.0.35 (stock global) to 2.28.62 using `miwifi_r4a_firmware_72d65_2.28.62.bin`, which worked
- Deciding to flash international firmware 3.0.24 using `miwifi_r4a_all_03233_3.0.24_INT.bin`, which finally bricked the router and I couldn't flash any firmware file at all via TFTP

I've read somewhere on the forum that flashing the original firmware would help solve the problem, but firmware 3.0.35 is not available, and cannot be extracted from another, properly working router. So, it looks to me that the only possible way of recovery is via SPI flashing device.

**Comment by thewh1teagle** [comment]
I just debricked mi 4a gigabit here are the notes I wrote while doing it

1. Download firmware from https://mirom.ezbox.idv.tw/en/miwifi/R4A/roms-stable/
2. Download debrick tool from https://hoddysguides.com/xiaomi-debrick-tools-all/
3. Connect network cable from the nearest port to the power to the PC
4. Place the bin file in the debrick tool folder
5. Open control panel and change ethernet adapter IP to 192.168.1.2 and subnet mask to 255.255.255.0
6. Delete the file test.bin from the debrick tool
7. Open debrick tool and set it to 192.168.1... instead of default one
8. Select the right file in debrick tool
9. Press online in debrick tool
10. Hold the reset button and reconned the power while watching the debrick log for changes
11. Wait for it to collect the file. it will show more log after DoReadFile once it finished.

Note: with Gigabit m4 it worked for me just with the test file that comes with the debrick tool
Note: It may take 20 minutes to install it. 
Note: Make sure your PC remain ON. You can use https://github.com/thewh1teagle/kipi
Note: 192.168.1.2 may missing from debrick tool. You may need to reopen it after doing step 10 again to see it
Note: Sometimes you need to long press the reset button AFTER it's ON instead of before to make the led flash and work. 
Note: sometimes after 20 minutes of flash it doesn't show something in the log or flash. Presonally I just reboot the router and notice that there's new WIFI available.


By the way I bricked it simply because I tried to install stock fw from openwrt settings. Coudln't revert it to stock.

**Comment by deathline94** [comment]
> I just debricked mi 4a gigabit here are the notes I wrote while doing it
> 
>     1. Download firmware from [mirom.ezbox.idv.tw/en/miwifi/R4A/roms-stable](https://mirom.ezbox.idv.tw/en/miwifi/R4A/roms-stable/)
> 
>     2. Download debrick tool from [hoddysguides.com/xiaomi-debrick-tools-all](https://hoddysguides.com/xiaomi-debrick-tools-all/)
> 
>     3. Connect network cable from the nearest port to the power to the PC
> 
>     4. Place the bin file in the debrick tool folder
> 
>     5. Open control panel and change ethernet adapter IP to 192.168.1.2 and subnet mask to 255.255.255.0
> 
>     6. Delete the file test.bin from the debrick tool
> 
>     7. Open debrick tool and set it to 192.168.1... instead of default one
> 
>     8. Select the right file in debrick tool
> 
>     9. Press online in debrick tool
> 
>     10. Hold the reset button and reconned the power while watching the debrick log for changes
> 
>     11. Wait for it to collect the file. it will show more log after DoReadFile once it finished.
> 
> 
> Note: with Gigabit m4 it worked for me just with the test file that comes with the debrick tool Note: It may take 20 minutes to install it. Note: Make sure your PC remain ON. You can use [thewh1teagle/kipi](https://github.com/thewh1teagle/kipi) Note: 192.168.1.2 may missing from debrick tool. You may need to reopen it after doing step 10 again to see it Note: Sometimes you need to long press the reset button AFTER it's ON instead of before to make the led flash and work. Note: sometimes after 20 minutes of flash it doesn't show something in the log or flash. Presonally I just reboot the router and notice that there's new WIFI available.
> 
> By the way I bricked it simply because I tried to install stock fw from openwrt settings. Coudln't revert it to stock.


That's pretty much a routine de-bricking for this model . But i don't know how or why, my 4A shipped with version 3.2.30
which doesn't exist anywhere apparently and sadly enough doesn't accept any other versions findable on the internet !

**Comment by deathline94** [comment]
> Got the same problem on another router 4A Gigabit Global version router, fast blinking purple light that blinks infinitely. I am going to order a CH341A programmer and flash the chip from another 4A Gigabit global router which has the exact same firmware, 3.0.35 and report here if I succeed
> 
> In my case the router was bricked by:
> 
>     * Flashing from 3.0.35 (stock global) to 2.28.62 using `miwifi_r4a_firmware_72d65_2.28.62.bin`, which worked
> 
>     * Deciding to flash international firmware 3.0.24 using `miwifi_r4a_all_03233_3.0.24_INT.bin`, which finally bricked the router and I couldn't flash any firmware file at all via TFTP
> 
> 
> I've read somewhere on the forum that flashing the original firmware would help solve the problem, but firmware 3.0.35 is not available, and cannot be extracted from another, properly working router. So, it looks to me that the only possible way of recovery is via SPI flashing device.

I'm curious as to what you ended up doing with yours . Any progress ?

**Comment by jrroshan** [comment]
> I just debricked mi 4a gigabit here are the notes I wrote while doing it
> 
> 1. Download firmware from https://mirom.ezbox.idv.tw/en/miwifi/R4A/roms-stable/
> 2. Download debrick tool from https://hoddysguides.com/xiaomi-debrick-tools-all/
> 3. Connect network cable from the nearest port to the power to the PC
> 4. Place the bin file in the debrick tool folder
> 5. Open control panel and change ethernet adapter IP to 192.168.1.2 and subnet mask to 255.255.255.0
> 6. Delete the file test.bin from the debrick tool
> 7. Open debrick tool and set it to 192.168.1... instead of default one
> 8. Select the right file in debrick tool
> 9. Press online in debrick tool
> 10. Hold the reset button and reconned the power while watching the debrick log for changes
> 11. Wait for it to collect the file. it will show more log after DoReadFile once it finished.
> 
> Note: with Gigabit m4 it worked for me just with the test file that comes with the debrick tool Note: It may take 20 minutes to install it. Note: Make sure your PC remain ON. You can use https://github.com/thewh1teagle/kipi Note: 192.168.1.2 may missing from debrick tool. You may need to reopen it after doing step 10 again to see it Note: Sometimes you need to long press the reset button AFTER it's ON instead of before to make the led flash and work. Note: sometimes after 20 minutes of flash it doesn't show something in the log or flash. Presonally I just reboot the router and notice that there's new WIFI available.
> 
> By the way I bricked it simply because I tried to install stock fw from openwrt settings. Coudln't revert it to stock.

when I tried this I have purple light blinking don't know what it mean

**Comment by DiiiPhuc** [comment]
i've flashed with a CH341A programmer after my router bricked
and it worked
this is stock rom i flashed
https://www.mediafire.com/file/o713ibzf5ailrr0/Xiaomi_4AG_Stock.7z/file

**Comment by CallMeMacadan** [comment]
> i've flashed with a CH341A programmer after my router bricked and it worked this is stock rom i flashed https://www.mediafire.com/file/o713ibzf5ailrr0/Xiaomi_4AG_Stock.7z/file

Did you have to "mod" your ch341a to 3.3v?

**Comment by DiiiPhuc** [comment]
> > i've flashed with a CH341A programmer after my router bricked and it worked this is stock rom i flashed https://www.mediafire.com/file/o713ibzf5ailrr0/Xiaomi_4AG_Stock.7z/file
> 
> Did you have to "mod" your ch341a to 3.3v?

No, 5v like stock

**Comment by CallMeMacadan** [comment]
> > > i've flashed with a CH341A programmer after my router bricked and it worked this is stock rom i flashed https://www.mediafire.com/file/o713ibzf5ailrr0/Xiaomi_4AG_Stock.7z/file
> > 
> > 
> > Did you have to "mod" your ch341a to 3.3v?
> 
> No, 5v like stock

What SPI was?

**Comment by sujankarki269** [comment]
> I just debricked mi 4a gigabit here are the notes I wrote while doing it
> 
> 1. Download firmware from https://mirom.ezbox.idv.tw/en/miwifi/R4A/roms-stable/
> 2. Download debrick tool from https://hoddysguides.com/xiaomi-debrick-tools-all/
> 3. Connect network cable from the nearest port to the power to the PC
> 4. Place the bin file in the debrick tool folder
> 5. Open control panel and change ethernet adapter IP to 192.168.1.2 and subnet mask to 255.255.255.0
> 6. Delete the file test.bin from the debrick tool
> 7. Open debrick tool and set it to 192.168.1... instead of default one
> 8. Select the right file in debrick tool
> 9. Press online in debrick tool
> 10. Hold the reset button and reconned the power while watching the debrick log for changes
> 11. Wait for it to collect the file. it will show more log after DoReadFile once it finished.
> 
> Note: with Gigabit m4 it worked for me just with the test file that comes with the debrick tool Note: It may take 20 minutes to install it. Note: Make sure your PC remain ON. You can use https://github.com/thewh1teagle/kipi Note: 192.168.1.2 may missing from debrick tool. You may need to reopen it after doing step 10 again to see it Note: Sometimes you need to long press the reset button AFTER it's ON instead of before to make the led flash and work. Note: sometimes after 20 minutes of flash it doesn't show something in the log or flash. Presonally I just reboot the router and notice that there's new WIFI available.
> 
> By the way I bricked it simply because I tried to install stock fw from openwrt settings. Coudln't revert it to stock.

THANKS Bro..You saved me...It works

**Comment by RickNyken** [comment]
> I just debricked mi 4a gigabit here are the notes I wrote while doing it
> 
> 1. Download firmware from https://mirom.ezbox.idv.tw/en/miwifi/R4A/roms-stable/
> 2. Download debrick tool from https://hoddysguides.com/xiaomi-debrick-tools-all/
> 3. Connect network cable from the nearest port to the power to the PC
> 4. Place the bin file in the debrick tool folder
> 5. Open control panel and change ethernet adapter IP to 192.168.1.2 and subnet mask to 255.255.255.0
> 6. Delete the file test.bin from the debrick tool
> 7. Open debrick tool and set it to 192.168.1... instead of default one
> 8. Select the right file in debrick tool
> 9. Press online in debrick tool
> 10. Hold the reset button and reconned the power while watching the debrick log for changes
> 11. Wait for it to collect the file. it will show more log after DoReadFile once it finished.
> 
> Note: with Gigabit m4 it worked for me just with the test file that comes with the debrick tool Note: It may take 20 minutes to install it. Note: Make sure your PC remain ON. You can use https://github.com/thewh1teagle/kipi Note: 192.168.1.2 may missing from debrick tool. You may need to reopen it after doing step 10 again to see it Note: Sometimes you need to long press the reset button AFTER it's ON instead of before to make the led flash and work. Note: sometimes after 20 minutes of flash it doesn't show something in the log or flash. Presonally I just reboot the router and notice that there's new WIFI available.
> 
> By the way I bricked it simply because I tried to install stock fw from openwrt settings. Coudln't revert it to stock.

OMG. My router bricked by itself 4 years ago, and I don't know why, but I never throw it away. Now I tried this and it was alive again. Thank you very much.

**Comment by george-walker** [comment]
> i've flashed with a CH341A programmer after my router bricked and it worked this is stock rom i flashed https://www.mediafire.com/file/o713ibzf5ailrr0/Xiaomi_4AG_Stock.7z/file

I had a CH341A mishap and ended up corrupted the flash chip.

The flash chip was a EN25QX128A (as per [IMSProg](https://github.com/bigbigmdm/IMSProg))

This image above worked for me by replacing the bootdelay value and following the instructions in https://forum.openwrt.org/t/xiaomi-mi-router-4a-gigabit-edition-r4ag-r4a-gigabit-fully-supported-and-flashable-with-openwrtinvasion/36685

Thanks for uploading @DiiiPhuc

**Comment by StefanStanic95** [comment]
> > Got the same problem on another router 4A Gigabit Global version router, fast blinking purple light that blinks infinitely. I am going to order a CH341A programmer and flash the chip from another 4A Gigabit global router which has the exact same firmware, 3.0.35 and report here if I succeed
> > In my case the router was bricked by:
> > ```
> > * Flashing from 3.0.35 (stock global) to 2.28.62 using `miwifi_r4a_firmware_72d65_2.28.62.bin`, which worked
> > 
> > * Deciding to flash international firmware 3.0.24 using `miwifi_r4a_all_03233_3.0.24_INT.bin`, which finally bricked the router and I couldn't flash any firmware file at all via TFTP
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I've read somewhere on the forum that flashing the original firmware would help solve the problem, but firmware 3.0.35 is not available, and cannot be extracted from another, properly working router. So, it looks to me that the only possible way of recovery is via SPI flashing device.
> 
> I'm curious as to what you ended up doing with yours . Any progress ?

Unfortunately I did not manage to recover the router. I bought a CH341A kit that has a 3.3V mod built-in, but I could not get it to read the flash properly from the chip, possibly due to bad contact or another issue.

**Comment by greensulinya** [comment]
> I just debricked mi 4a gigabit here are the notes I wrote while doing it
> 
> 1. Download firmware from https://mirom.ezbox.idv.tw/en/miwifi/R4A/roms-stable/
> 2. Download debrick tool from https://hoddysguides.com/xiaomi-debrick-tools-all/
> 3. Connect network cable from the nearest port to the power to the PC
> 4. Place the bin file in the debrick tool folder
> 5. Open control panel and change ethernet adapter IP to 192.168.1.2 and subnet mask to 255.255.255.0
> 6. Delete the file test.bin from the debrick tool
> 7. Open debrick tool and set it to 192.168.1... instead of default one
> 8. Select the right file in debrick tool
> 9. Press online in debrick tool
> 10. Hold the reset button and reconned the power while watching the debrick log for changes
> 11. Wait for it to collect the file. it will show more log after DoReadFile once it finished.
> 
> Note: with Gigabit m4 it worked for me just with the test file that comes with the debrick tool Note: It may take 20 minutes to install it. Note: Make sure your PC remain ON. You can use https://github.com/thewh1teagle/kipi Note: 192.168.1.2 may missing from debrick tool. You may need to reopen it after doing step 10 again to see it Note: Sometimes you need to long press the reset button AFTER it's ON instead of before to make the led flash and work. Note: sometimes after 20 minutes of flash it doesn't show something in the log or flash. Presonally I just reboot the router and notice that there's new WIFI available.
> 
> By the way I bricked it simply because I tried to install stock fw from openwrt settings. Coudln't revert it to stock.

it helped me in this order - after step 8, you need to turn off the power to the router, then 9 and in order. the firmware is needed under the name "miwifi_r4av2_firmware_6bdd4_2.30.500.bin" from the website https://www1.miwifi.com/ . I did not rename the firmware.

**Comment by vsramadanov** [comment]
Hello! I've got a router labeled with R4AC (does C stand for Chinese?) and no firmware seems to fit me. Could you please help me to find suitable firmware?
P.S. I use a linux machine as a host + dnsmasq + tftpd-hpa to transfer the firmware. Does the debrick tool from hobbiesguides do something more then just send FW to the router?

**Comment by deathline94** [comment]
> Hello! I've got a router labeled with R4AC (does C stand for Chinese?) and no firmware seems to fit me. Could you please help me to find suitable firmware? P.S. I use a linux machine as a host + dnsmasq + tftpd-hpa to transfer the firmware. **Does the debrick tool from hobbiesguides do something more then just send FW to the router?**

naah pretty much a payload sender .also 1.5 years has passed and i havent found my firmware yet . my router was shipped with the 3.2.30 firmware
